### ISSUE TITLE
bindist rehash

### ISSUE THREAD

**Issue by isaacs** [npm/npm#1917]
New plan for binary distribution future.

1. The `--bindist` config flag shall be `<node version>-<os.platform()>-<os.arch()>` and that's it.

    Presumably just the version family is enough for "stable" releases, but if we're going to get away from the even/odd thing, then maybe it makes sense to just always use major/minor for release versions, and the full version when publishing with a -pre node.  In either case, it should be treated as a semver range, so a full or partial match will work on publishing.

2. Re-enable bin-publish by default, but only for packages that have a .gyp file in them.

    I think we'll probably need to have some other way to make sure that it's a gyp file for building a node addon, and not just someone using gyp for some other purpose.

3. Drop automatic wscript detection in npm 1.2.  @ry suggests removing the "install" script hook altogether, but that strikes @isaacs as overly excessive in some ways and not excessive enough in others.

More discussion/ranting here: http://piscisaureus.no.de/log/2011-12-21

**Comment by othiym23** [comment]
This discussion has been superseded by one in the Node.js build WG: https://github.com/nodejs/build/issues/151. Somebody from the CLI team, probably me, needs to help get @orangemocha and the rest of the interested parties there unblocked; it's been on my todo list for a few months, but 2016 has been hectic so far. Coming up with a reliable, scalable, supportable way to replace the need to do native builds locally is even more of a pressing issue now than before, given the large scale of npm use on Windows and the still-too-complex process of getting a solid native build toolchain bootstrapped on Windows. Thanks for capturing this note, and I'm hopeful that in the near future, the community as a whole will come up with a solid way to address this.
